pos_order_reorder/static/src/js/Screens/TicketScreen/ControlButtons/ReorderButton.esm.js

```diff
@@ -0,0 +1,67 @@
+/** @odoo-module **/
+
+import PosComponent from "point_of_sale.PosComponent";
+import Registries from "point_of_sale.Registries";
+import {Orderline} from "point_of_sale.models";
+
+class ReorderButton extends PosComponent {
+    get isEmptyOrder() {
+        if (!this.props.order) return true;
+        return this.props.order.is_empty();
+    }
+    _reOrder() {
+        if (this.isEmptyOrder) {
+            return;
+        }
+        const order = this.props.order;
+        const pos = this.env.pos;
+        const partner = order.get_partner();
+        const newOrder = pos.add_new_order();
+        if (partner) {
+            newOrder.set_partner(partner);
+        }
+        if (order.fiscal_position) {
+            newOrder.fiscal_position = order.fiscal_position;
+        }
+        if (order.pricelist) {
+            newOrder.set_pricelist(order.pricelist);
+        }
+        const lines = order.get_orderlines();
+        for (var i = 0; i < lines.length; i++) {
+            const line = lines[i];
+            const new_line = Orderline.create(
+                {},
+                {
+                    pos: pos,
+                    order: newOrder,
+                    product: pos.db.get_product_by_id(line.get_product().id),
+                    description: line.name,
+                    price: line.price_unit,
+                    tax_ids: newOrder.fiscal_position ? undefined : line.tax_id,
+                    price_manually_set: true,
+                    customer_note: line.customer_note,
+                }
```

Comment:
Move this vals preparation to another method, for example: `_prepareReorderLineVals` or similar
This way we have a hook point for other modules to add their own fields/properties